ament_cmake_core/ament_cmake_environment-extras.cmake

```diff
@@ -14,11 +14,10 @@
 
 # copied from ament_cmake_core/ament_cmake_environment-extras.cmake
 
-option(AMENT_CMAKE_ENVIRONMENT_GENERATION
-  "Generate environment files in the CMAKE_INSTALL_PREFIX" OFF)
 option(AMENT_CMAKE_ENVIRONMENT_PARENT_PREFIX_PATH_GENERATION
   "Generate marker file containing the parent prefix path" ON)
 
+# This is used by ros_workspace
```

Comment:
If this comment needs to be here (there's no other ament_cmake_core functions that have to justify their inclusion), I think it would make sense to describe the available functionality rather than list what utilizes it. ros_workspace could change or other platforms or build tools may come along and use this function to provide these scripts as ament_tools once did. Since ament_generate_environment is documented in its source file and is the only content of that file perhaps the function definition should just be inlined instead?

Reply:
> If this comment needs to be here (there's no other ament_cmake_core functions that have to justify their inclusion), I think it would make sense to describe the available functionality rather than list what utilizes it. 

Yeah, no other comment says what's using this downstream. I think it's a clue into my thinking.

> ros_workspace could change or other platforms or build tools may come along and use this function to provide these scripts as ament_tools once did. Since ament_generate_environment is documented in its source file and is the only content of that file perhaps the function definition should just be inlined instead?

IIUC the prefix level scripts are defining the idea of a workspace of binary packages, and a single CMake package using `ament_cmake` doesn't need to know how to create that. I'm not sure what future build tools will need, but the only build tool being used now, colcon, has its own ideas about what a workspace is and provides its own scripts. How about we move the `prefix_level` scripts and this helper function into `ros_workspace` (assuming it's the only place that uses it)? That way the idea of how to create a workspace is kept in one place.

Reply:
I opened a PR for discussion about moving the prefix_level scripts to `ros_workspace`. https://github.com/ros2/ros_workspace/pull/22

Reply:
I removed this comment in 5428cc0